### ISSUE TITLE
Refactor API to be more pythonic

### ISSUE THREAD

**Issue by liampauling** [tarb/betfair_data#5]
Is there any scope to refactor the API/ReadMe so that its use is more pythonic, for example currently it is tricky to read / understand what and where it is doing the processing:

```python
for file in bfd.Files(paths).iter():
    market_count += 1
    for market in file:
        update_count += 1
```

Small change to declare the files first before iterating:

```python
files = bfd.Files(paths)

for file in files.iter():
    market_count += 1
    for market in file:
        update_count += 1
```

This could be improved further by utilising python's builtins `__iter__` removing the requirement to call `iter` entirely.

```python
files = bfd.Files(paths)

for file in files:
    market_count += 1
    for market in file:
        update_count += 1
```

bflw processing can stay the same

```python
files = bfd.Files(paths)

for file in files.bflw():
    market_count += 1
    for market in file:
        update_count += 1
```

Is there any reason why the mutable flag isn't available at the Files level?

**Comment by tarb** [comment]
Hey, I've actually gone backwards and forwards on this a few times already and I've never been really happy with it. 
The challenge is exposing different options between the different outputs. Ie, no mutable implementation for bflw, and soon stream_id on bflw.
What I'm currently leaning towards is just exposing a different Files constructor under the bflw package, so we could have
```py
files = bfd.Files(paths, mutable=True)

# or for bflw

files = bflw.Files(paths)
```

Then i can just put the implementation specific parameters on their own Files constructor, and completely integrate the iterator onto that object and everything gets a bit simpler. The downside is it could be weird to have both a bfd.Files object and a bflw.Files object in one codebase, but I doubt this would actually be a common occurrence outside testing scripts

**Comment by liampauling** [comment]
Yeah I think that is fine, I see this library becoming a 'core' library with (hopefully) lots sitting on top which will do their own abstraction as required.

Maybe it's the naming of `Files` that I find confusing, as it's a bit more than that.

**Comment by tarb** [comment]
At its core Files is just a loop calling and returning the File constructor. It's supposed to reduce the barrier of entry so people dont have to worry about knowing how to read files / decompress / open archives, they can just get going with one function and a set of paths. (The File constructor needs the bytes to be decompressed)
The performance benefits of it running in a seperate rust thread are completely secondary, and Im completely convinced a better python programmer than me would be able to wrangle either asyncio / python threads / multiprocessing to achieve the same speed.
So my reasoning is Files == loop of File, handling the common file storage cases

**Comment by tarb** [comment]
Closed in #6
